### DISCUSSION TITLE
Generating a Preview Image PrusaSlicer for Mac post-processing script issues! Please Help :)

### DISCUSSION THREAD

**Discussion by Vysper7** [mriscoc/Ender3V2S1#588] [Q&A]
I am trying to generate preview images for my Ender 3 S1 machine and am running PrusaSlicer on my Mac. My issue seems to be that I cannot get the post-processing to work. It really likes to send me this error:
<img width="691" alt="Screenshot 2022-12-30 at 9 44 36 PM" src="https://user-images.githubusercontent.com/90841957/210098468-e872f1e9-70bc-40ad-859c-8d558ce54b1b.png">
This is the post processing script I use:
`/usr/bin/python3 /Users/agentvinci/Library/Application\Support/PrusaSlicer/Scripts/CreateHeaderAndThumbnail.py;`
I have also tried variations like this:
`/usr/bin/python3\ /Users/agentvinci/Library/Application\Support/PrusaSlicer/Scripts/CreateHeaderAndThumbnail.py;`
Or:
`"/usr/bin/python3" "/Users/agentvinci/Library/Application\Support/PrusaSlicer/Scripts/CreateHeaderAndThumbnail.py"`
But I can never get it to work.
I have python 3 installed on my Mac.
I have the Pillow library for python installed.
My .py script is correctly stored in PrusaSlicers Configuration Folder/Scripts
It has send me different errors in the past such as 126 or 127 but I think I got them sorted.
Please help
Thanks you so much,
Agent Vinci

**Comment by mriscoc** [comment]
**Application\Support**  maybe should be **Application/Support**

**Reply by Vysper7** [reply]
Thank you for your suggestion but the folder is called 'Application Support' with a space between the words. I have found out the PrusaSlicer doesn't understand spaces and instead you have to show them with a '\' character. Without this we get the 127 or 126 error I think.

**Reply by mriscoc** [reply]
Error 2 means that Python cannot find the script in the specified path. You can see in the error response that there isn't a space between Application and Support.

**Reply by Vysper7** [reply]
<img width="894" alt="Screenshot 2022-12-30 at 10 50 44 PM" src="https://user-images.githubusercontent.com/90841957/210103254-84b759be-6946-408a-82e3-53fd1aff9202.png">
Ok I tried that and it worked! Now we have a different error though. It's saying the image results are too large and it cannot find a module named 'PIL'.

**Reply by mriscoc** [reply]
I don't have a MAC but it seems related to a missing library in your Python installation.
Check: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview#mac-advices

**Reply by Vysper7** [reply]
Thank you for the suggestion and your help. The Mac has python 3 installed with all the libraries. I have successfully used cura to generate thumbnail images. However, it turns out that the screen of the Ender 3 S1 doesn't support thumbnail images anyway if you have a DACAI screen (with was mentioned in the article) and so it turns out that I have a DACAI screen :(. PrusaSlicer still doesn't generate images but it doesn't really matter anymore since I switched back to the maybe slightly inferior but properly configured Creality Slicer (Cura skin) and am printing successfully though it. Thanks again for your support.
